gammapy/scripts/analysis.py

```diff
@@ -63,13 +64,16 @@ def __init__(self, config=None, template="basic"):
         self.observations = None
         self.geom = None
```

Comment:
The information such as `geom`, `psf_kernel`, `maps`, `images`  is available on the `Dataset` objects, so users can write:
```
analysis.datasets.datasets[0].psf
analysis.datasets.datasets[0].counts.geom
```
I realise, that this is not as convenient as having it directly on the `Analysis` object, but the problem with these single arguments is, that in the end there will be many datasets. I.e. not only a single `PSFKernel` but 20 of those. The same goes for the the `geom` in the end there are many (different) geoms like for the counts map, the exposure map and IRF maps.

My proposal would be to remove `self.geom`, `self.psf_kernel`, `self.maps` and `self.images`. Instead I could  open a follow-up PR to simplify the dataset access so that users can access datasets by name like so:
```
analysis.datasets["obs-1234"].psf
analysis.datasets["obs-1234"].counts.geom
```
What do you think?